### PR TITLE
Taskdefinition template

### PR DESCRIPTION
**Pull request by dennari**
Can now restart a service with a taskDefinition "template" that is fetched from S3 and updated with the latest image tag. The command is `ecs-upd8r -s restart-terraform`.

### CODE REVIEW

File: src/ecs-deploy.ts [hunk 1]
```diff
@@ -25,73 +135,167 @@ export function deploy(opts: IConfig, onlyTask?: boolean) {
...
-  console.log('Next image: ' + nextImage + ':' + nextTag);
-
   return _.assign({}, container, {
     image: nextImage + ':' + nextTag,
   });
 }
 
+export async function syncRevision(config: IConfig, taskDefinition: RegisteredTaskDefinition) {
+  if (!config.BUCKET || !config.KEY) {
+    throw new Error(`Can't syncRevision without since BUCKET or KEY is not set`);
```
Comment (juhoojala) [inline src/ecs-deploy.ts]: without what?

File: src/index.ts [hunk 2]
```diff
@@ -9,19 +9,31 @@ import config, { IConfig } from './config';
 import * as deployer from './ecs-deploy';
 import { promisify } from './promisify';
 
+let version = 'unknown';
+try {
+  version = require('../../package').version;
```
Comment (juhoojala) [inline src/index.ts]: Could use https://www.npmjs.com/package/pkginfo

File: src/index.ts [hunk 3]
```diff
@@ -63,6 +79,35 @@ function fail(msg?: string) {
...
+  };
+}
+
+export async function terraformRestart(config: IConfig) {
+
+  console.log(chalk.bold.green('\nRestarting the service to use the latest taskDefinition in S3\n'));
+
+  const {container, taskDefinition} = await deployer.terraformRestart(config);
+
+  console.log('Updated service %s to use image %s', config.SERVICE, container.image);
```
Comment (juhoojala) [inline src/index.ts]: Is this log message correct?

**Comment by juhoojala** [comment]
Running `tsc` or (`npm run-script prepublish`) yields:

```
4 import config from './config.test';

test/tests.ts(4,20): error TS2307: Cannot find module './config.test'.
```

**Comment by juhoojala** [comment]
Since there are now tests, can we run them in circle?

**Comment by dennari** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;EdYO6qUR&#x2F;123-terraform-tweaks"> Terraform tweaks</a>

**Comment by juhoojala** [comment]
`restart-terraform` subcommand is missing from usage output. obviously a minor issue.

**Comment by juhoojala** [comment]
Otherwise looks ok.

**Comment by dennari** [comment]
The README leaves a lot to be desired and the naming of the commands is definitely suboptimal, but I'll improve those later.

**Pull request merged**